### PR TITLE
First PR : Coleman-back-office

### PR DESCRIPTION
**Pull request by Samuel-Corcaud-KeyConsulting**

### CODE REVIEW

**Review by nicolasTurban — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,175 @@
...
+#Keycloak configuration
+keycloak.realm=insee-realm
+keycloak.resource=coleman
+keycloak.auth-server-url=http://localhost:8180/auth
+keycloak.public-client=true
+keycloak.bearer-only=true
+keycloak.principal-attribute:preferred_username
+
+coleman.environnement="environnement local"
+fr.insee.coleman.ldap.services.uri.hote=https://contacts.insee.test
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply README.md]: done

File: src/main/java/fr/insee/coleman/api/DataLoader.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+			orderService.saveAndFlush(new Order(Long.parseLong("6"), TypeManagementMonitoringInfo.VALPAP.toString(), 6));
+			orderService.saveAndFlush(new Order(Long.parseLong("5"), TypeManagementMonitoringInfo.HC.toString(), 5));
+			orderService.saveAndFlush(new Order(Long.parseLong("4"), TypeManagementMonitoringInfo.PARTIELINT.toString(), 4));
+			orderService.saveAndFlush(new Order(Long.parseLong("3"), TypeManagementMonitoringInfo.WASTE.toString(), 3));
+			orderService.saveAndFlush(new Order(Long.parseLong("2"), TypeManagementMonitoringInfo.PND.toString(), 2));
+			orderService.saveAndFlush(new Order(Long.parseLong("1"), TypeManagementMonitoringInfo.INITLA.toString(), 1));
+	
+			// Creating campaigns
+			campaignService.save(
+					new Campaign("simpsons2021x00","Survey on the Simpsons tv show 2021", 1605909600000l, 1640995199000l));
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/main/java/fr/insee/coleman/api/DataLoader.java]: done

File: src/main/java/fr/insee/coleman/api/controller/RestManagementMonitoringInfo.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
+package fr.insee.coleman.api.controller;
+
+import java.util.List;
+
+//TODO Check if needed
+//import org.apache.velocity.exception.ResourceNotFoundException;
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/main/java/fr/insee/coleman/api/controller/RestManagementMonitoringInfo.java]: done

File: src/main/java/fr/insee/coleman/api/controller/RestManagementMonitoringInfo.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	@GetMapping(value = "/campaigns/{idCampaign}/survey-units/{idSu}/management-monitoring-infos", produces = "application/json")
+	public JSONCollectionWrapper<ManagementMonitoringInfo> displayManagementMonitoringInfoOfOneSurveyUnit(@PathVariable String idSu,
+			@PathVariable String idCampaign) {
+		LOGGER.info("Request GET to display the folowwing informations for the survey-unit n° {}, campaign n° {}", idSu, idCampaign);
+
+		return new JSONCollectionWrapper<ManagementMonitoringInfo>(
+				managementMonitoringInfoService.findBySurveyUnit(surveyUnitService.findByIdSurveyUnitAndIdCampaign(idSu, idCampaign)));
+	}
+
+	// Vérifier les isg que l'on peut delete ou non
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/main/java/fr/insee/coleman/api/controller/RestManagementMonitoringInfo.java]: done

File: src/main/java/fr/insee/coleman/api/services/UploadService.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+	}
+
+	public Upload saveAndFlush(Upload up) {
+		return uploadRepository.saveAndFlush(up);
+	}
+
+	public void delete(Upload up) {
+		uploadRepository.delete(up);
+	}
+	// Checks that the ISG of an upload belong to the same campaign
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/main/java/fr/insee/coleman/api/services/UploadService.java]: done

File: src/test/java/fr/insee/coleman/api/RestAuthBasicTest.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+	 * @throws Exception
+	 * 
+	 * @throws JSONException
+	 */
+	@Test
+	@Order(1)
+	void testFindCampaign() throws Exception {
+		RestAssured.port = port;
+		given().auth().preemptive().basic("ABC", "abc").when().get("/campaigns").then().statusCode(200).and().assertThat()
+				.body("datas[0].id", equalTo("simpsons2021x00"));
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/test/java/fr/insee/coleman/api/RestAuthBasicTest.java]: done

File: src/test/java/fr/insee/coleman/api/services/ManagementMonitoringInfoServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+	public void testSaveIhmError() throws Exception {
+		String idCampagne="toto";
+		ManagementMonitoringInfoDto managementMonitoringInfoDto = new ManagementMonitoringInfoDto();
+		managementMonitoringInfoDto.setIdSu("A00000789");
+		managementMonitoringInfoDto.setStatus(TypeManagementMonitoringInfo.FOLLOWUP);
+
+		Upload upload = new Upload();
+		upload.setDate((long) 100);
+
+//		Mockito.when(ueService.findByIdUe("A00000789")).thenReturn(null);
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/test/java/fr/insee/coleman/api/services/ManagementMonitoringInfoServiceTest.java:57]: done

File: src/test/java/fr/insee/coleman/api/services/ManagementMonitoringInfoServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+		assertThrows(RessourceNotFoundException.class, () -> {managementMonitoringInfoService.saveIhm(idCampagne, managementMonitoringInfoDto, upload);});
+	}
+
+	@Test
+	public void testsaveBatchErrorSu() throws Exception {
+		ManagementMonitoringInfoDto managementMonitoringInfoDto = new ManagementMonitoringInfoDto();
+		managementMonitoringInfoDto.setIdSu("B00000789");
+		managementMonitoringInfoDto.setStatus(TypeManagementMonitoringInfo.PARTIELINT);
+
+		//Mockito.when(ueService.findByIdUe("B00000789")).thenReturn(null);
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/test/java/fr/insee/coleman/api/services/ManagementMonitoringInfoServiceTest.java:68]: done

File: src/test/java/fr/insee/coleman/api/services/SurveyUnitServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+//		suTest.setManagementMonitoringInfos(managementMonitoringInfoList);
+//
+//		Mockito.when(managementMonitoringInfoService.findBySurveyUnit(suTest)).thenReturn(suTest.getManagementMonitoringInfos());
+//
+//		Boolean actual = ueService.estARelancer(suTest);
+//		
+//		System.out.println(managementMonitoringInfoList.toString());
+//
+//		Assert.assertEquals(false, actual);
+//	}
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/test/java/fr/insee/coleman/api/services/SurveyUnitServiceTest.java]: done

File: src/main/java/fr/insee/coleman/api/services/UploadService.java [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+		delete(up);
+		return true;
+	}
+
+	public Optional<Upload> findById(long id) {
+		return uploadRepository.findById(id);
+	}
+
+	public List<Upload> findAllByIdCampaign(String idCampaign) {
+		// Keeps the uploads which first ISQ belongs to the survey
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/main/java/fr/insee/coleman/api/services/UploadService.java]: done

File: src/main/java/fr/insee/coleman/api/services/SurveyUnitService.java [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+		Date date = new Date();
+		managementMonitoringInfoService.saveAndFlush(new ManagementMonitoringInfo(null, suSearched, TypeManagementMonitoringInfo.INITLA, date.getTime(), null));
+	}
+
+	public Collection<ExtractionRow> getSurveyUnitsToFollowUp(String idCampaign) {
+		return surveyUnitRepository.getSurveyUnitToFollowUp(idCampaign);
+	}
+
+	public TypeManagementMonitoringInfo determineState(SurveyUnit su) {
+		HashSet<TypeManagementMonitoringInfo> etats = new HashSet<TypeManagementMonitoringInfo>();
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/main/java/fr/insee/coleman/api/services/SurveyUnitService.java]: done

File: src/main/java/fr/insee/coleman/api/services/ManagementMonitoringInfoService.java [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+
+	public ResultUpload saveManagementMonitoringInfoInMassProcessingBatch(String idCampaign, List<ManagementMonitoringInfoDto> request) {
+
+		ResultUpload result = new ResultUpload();
+
+		for (ManagementMonitoringInfoDto mm : request) {
+			String id = mm.getIdSu() != null ? mm.getIdSu() : mm.getIdContact();
+			try {
+				saveBatch(idCampaign, mm);
+				result.ajouterIdOk(id);
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/main/java/fr/insee/coleman/api/services/ManagementMonitoringInfoService.java]: done

File: src/main/java/fr/insee/coleman/api/domain/SurveyUnit.java [hunk 13]
```diff
@@ -0,0 +1,165 @@
...
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonProperty.Access;
+import com.fasterxml.jackson.annotation.JsonRootName;
+
+@Entity
+@Table(name = "survey_unit")
+// @EqualsAndHashCode(onlyExplicitlyIncluded = true)
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/main/java/fr/insee/coleman/api/domain/SurveyUnit.java]: done

File: src/main/java/fr/insee/coleman/api/ColemanBackOfficeApplication.java [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+	@Override
+	protected SpringApplicationBuilder configure(SpringApplicationBuilder application) {
+		setProperties(); 
+		return application.sources(ColemanBackOfficeApplication.class);
+	}
+	
+	public static void setProperties() {
+		System.setProperty("spring.config.location",
+				"classpath:/,"
+        + "file:///${catalina.base}/webapps/colemanbo.properties");
```
[the remark this answers is not in the document]
Comment (kc-bclaudel) [reply src/main/java/fr/insee/coleman/api/ColemanBackOfficeApplication.java]: done

File: pom.xml [hunk 15]
```diff
@@ -0,0 +1,285 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>fr.insee.coleman.api</groupId>
+	<artifactId>coleman-back-office</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
```
Comment (nicolasTurban) [inline pom.xml:7]: Set to 0.9.0

**Review by nicolasTurban — CHANGES_REQUESTED** [review]

**Pull request merged**